engine/src/test/java/org/pentaho/di/trans/DefaultTransFactoryTest.java

```diff
@@ -0,0 +1,33 @@
+/*! ******************************************************************************
+ *
+ * Pentaho
+ *
+ * Copyright (C) 2024 by Hitachi Vantara, LLC : http://www.pentaho.com
+ *
+ * Use of this software is governed by the Business Source License included
+ * in the LICENSE.TXT file.
+ *
+ * Change Date: 2029-07-20
+ ******************************************************************************/
+
+package org.pentaho.di.trans;
+
+import org.junit.Test;
+import org.pentaho.di.core.exception.KettleException;
+import org.pentaho.di.core.logging.LoggingObject;
+import org.pentaho.di.core.logging.LoggingObjectInterface;
+import org.pentaho.di.core.util.Assert;
+import org.pentaho.di.core.variables.Variables;
+
+public class DefaultTransFactoryTest {
+
+
+  @Test
+  public void testCreate() throws KettleException {
+    TransMeta meta =
```

Comment:
Try using a mock next time, so complex class constructors won't fail your test.

```TransMeta meta = Mockito.mock(TransMeta.class)```

This will work, but next time consider using a mock due to out TransMeta Class might not be a simple class. And if the Trans or any other object has actually complex logic during constructor that require configuration like filesystem configuration or IO then your test could fail.

Reply:
I initially used mock. Then it started failing on a few other items used in the method call. And had to declare a lot of other mocks for that. I found this example used in on of the other test cases and used it in a similar way